frontend/src/pages/ActivityReport/index.js

```diff
@@ -94,7 +95,7 @@ function ActivityReport({ match, user, location }) {
         updateLoading(true);
 
         const apiCalls = [
-          getRecipients(),
+          getRecipients(allRegionsUserHasPermissionTo(user)),
```

Comment:
Reports are tied to a single region, so I think we need a small rewrite here. We probably want to retrieve the report from the API first and then use the region defined on the report to grab all the other data (collaborators, recipients and approvers).  We'd need the regionId passed into the report component for new reports (maybe as a query param? maybe state?). I _think_ @kryswisnaskas was working on creating reports in specific regions, she might have some input.

Reply:
Thinking about it a bit I think for new reports it is probably best to figure out the region in this component instead of having it passed in. That way any way no matter how you get to the `new` reports page you will have a region set.

Reply:
> We'd need the regionId passed into the report component for new reports (maybe as a query param? maybe state?). 

Yes, I think this makes sense. Since each report is associated with a single region, that region could be passed in. Right now we are limiting one region per user, but I can see post MVP we would need to support users having read permissions to multiple regions in which case the UI could give them a choice.

Reply:
> Thinking about it a bit I think for new reports it is probably best to figure out the region in this component instead of having it passed in. That way any way no matter how you get to the new reports page you will have a region set.

This is a good point. Do you anticipate this will be a case for the MVP though?

We could use the "pass in" approach since that appears to be the main way users get to create new reports, but also think about the need to have the new report component figure out the region in case there is no report id passed in.

The logic in this case would be:
- use the passed in regionId
- if no regionId was passed, use a fallback function to figure out the region.

Reply:
What would be the logic for figuring out which region this report should belong to in case no region was passed in?

Reply:
If the user is set up with read/write permission to one region, we could query user's permissions and find the region. 

Things get complicated if there were multiple read/write regions. I don't think that's a use case we need to worry about now though.

Reply:
In https://github.com/adhocteam/Head-Start-TTADP/pull/182
there is already a function checking for read/write. We could modify it to return a region.

Reply:
Updated to have that logic. Question, will there will never be a scenario where the user doesn't have read/writes to any region right?

Reply:
The way I have the code is to follow this flow:
1. If a `region` was passed in use that.
2. if none was passed in, calculate what region the user has read/write access  and return the first region that matches that. (We should ignore all other permissions such as READ only right? I feel like we should also include ADMIN rights in this search)
3. If for some reason, no region was found, we will have a weird state, where the region is -1 (since that is what we return when we don't find a region that matches). I want to prevent this state since the app will not work properly, but from what I can gather, we are guaranteed that this scenario won't happen. Please correct if i'm wrong

Reply:
> 3. If for some reason, no region was found, we will have a weird state, where the region is -1 (since that is what we return when we don't find a region that matches). I want to prevent this state since the app will not work properly, but from what I can gather, we are guaranteed that this scenario won't happen. Please correct if i'm wrong

We still need to handle this case, there is nothing preventing a user from having no write permissions, it all depends on how the user is setup. A redirect to the frontend would probably work for now. That would hopefully prompt the user to ask why they can't fill out a report which will lead to the us fixing their permission issue. In the future we will want to show a message of some kind on the redirect letting the user know they cannot create reports.